test/docker.js

```diff
@@ -507,4 +507,78 @@ describe("#docker", function() {
       docker.info(handler);
     });
   });
+
+  describe("#labelsAndFilters", function() {
+
+    var test_container;
+    var test_container_name = 'dockerode_test_querystring';
+    var labels = {
+      "dockerode-test": "",
+      "dockerode-test-value": "works"
+    };
+
+    // cleanup querystring container after tests
+    after(function(done){
+      if (!test_container) return done();
+      test_container.remove(function(err, data) {
+        return done(err);
+      });
+    });
+
+
+    it("should create a container from a map of labels", function(done){
+      this.timeout(5000);
+
+      function handler(err, container) {
+        expect(err).to.be.null;
+        expect(container).to.be.ok;
+        test_container = container;
+        done();
+      };
+
+      docker.createContainer({
+        "Image": testImage,
+        "Cmd": ['/bin/bash'],
+        "Labels": labels,
+        "name": test_container_name
+      }, handler);
+
+    });
+
+    it("should have the same labels used to create it", function(done){
+      test_container.inspect(function(err, info) {
+        expect(err).to.be.null;
+        expect(info.Name).to.equal('/' + test_container_name);
+        expect(info.Config.Labels).to.deep.equal(labels);
+        done();
+      });
+    });
+
+    it("should appear in listings filtering by a string of labels", function(done){
+      docker.listContainers({
+        "limit": 3,
+        "filters": '{"label": ["dockerode-test", "dockerode-test-value=works"]}'
+      }, function(err, data){
+        expect(data.length).to.equal(1);
```

Comment:
@rmg good catch. it won't if it's the only existing container && I was worried about this as well. it's tricky, as sometimes filters return all containers, particularly if the value is truncated upstream at docker-modem. 

one thought is to create two containers, one without the key=value pair?